coprocessor/fhevm-engine/fhevm-listener/src/health_check.rs

```diff
@@ -1,247 +1,108 @@
-        if elapsed_time > 30 {
-            self.message += &format!("Last tick is too old {elapsed_time}.\n");
-            self.unhealthy();
-        }
-    }
-
-    pub async fn check_database_connected(&mut self, database_url: &str) {
-        let Ok(options) = database_url.parse::<PgConnectOptions>() else {
-            self.message += "Bad Database url.\n";
-            self.database_connected = false;
-            self.unhealthy();
-            return;
-        };
-        let pool = PgPoolOptions::new()
-            .max_connections(1)
-            .connect_with(options.clone())
-            .await;
-        if let Err(_err) = pool {
-            self.message += "Database connection failed.\n";
-            self.database_connected = false;
-            self.unhealthy();
+    pub async fn update(&self) {
+        let now = SystemTime::now().duration_since(UNIX_EPOCH);
+        let Ok(now) = now else {
             return;
         };
-        self.database_connected = true;
+        *self.0.write().await = now.as_secs();
     }
 
-    pub async fn check_blockchain_connected(&mut self, blockchain_url: &str) {
-        let ws = WsConnect::new(blockchain_url);
-        let provider = ProviderBuilder::new().connect_ws(ws).await;
-        if let Err(_err) = provider {
-            self.message += "Blockchain connection failed.\n";
-            self.blockchain_connected = false;
-            self.unhealthy();
-            return;
+    pub async fn is_recent(&self, seconds: u64) -> bool {
+        let now = SystemTime::now().duration_since(UNIX_EPOCH);
+        let Ok(now) = now else {
+            return false;
         };
-        self.blockchain_connected = true;
+        let tick = self.0.read().await;
+        now.as_secs() - *tick < seconds
     }
 }
 
-#[derive(Clone)]
-pub struct HealthStateContent {
-    pub status: Health,
-    database_url: String,
-    blockchain_url: String,
-}
-
-impl HealthStateContent {
-    pub fn tick(&mut self) {
-        self.status.tick();
+impl Default for Tick {
+    fn default() -> Self {
+        Self::new()
     }
 }
 
-pub type HealthState = Arc<RwLock<HealthStateContent>>;
-
-#[derive(Clone)]
-pub struct HealthCheck {
-    pub health_state: HealthState,
-    port: u16,
-    pub cancel_token: CancellationToken,
-}
-
-impl HealthCheck {
-    pub fn new(
-        port: u16,
-        cancel_token: CancellationToken,
-        database_url: &str,
-        blockchain_url: &str,
-    ) -> Self {
-        let health_state = HealthStateContent {
-            status: Health::initial(),
-            database_url: database_url.to_owned(),
-            blockchain_url: blockchain_url.to_owned(),
+impl HealthCheckService for HealthCheck {
+    async fn health_check(&self) -> HealthStatus {
+        let mut status = HealthStatus::default();
+        // service inner loop
+        let check_alive = self.is_alive().await;
+        status.set_custom_check("alive", check_alive, false);
```

Comment:
Why do we need an extra custom check `alive`? 

Instead, you should only populate the endpoint `/liveness`

Reply:
Why would the service says "healthy" when it's blocked ?

Reply:
We should call `is_alive()`  when populating the `/healthz` status instead. Anyway, let's do this with another PR